saleor/checkout/utils.py

```diff
@@ -902,7 +902,11 @@ def ready_to_place_order(cart: Cart, taxes, discounts):
             return False, pgettext_lazy(
                 'order placement error',
                 'Shipping method is not valid for your shipping address')
-    if not is_fully_paid(cart, taxes, discounts):
+    if not cart.billing_address:
         return False, pgettext_lazy(
-            'order placement error', 'Checkout is not fully paid')
+            'order placement_error', 'Billing address is not set')
+    if not can_be_fully_paid(cart, taxes, discounts):
+        return False, pgettext_lazy(
+            'order placement error',
+            'Checkout can not be fully paid with proper payments')
```

Comment:
Similarly, for this error message, I'd say something like: "Provided amount is smaller than checkout's total". Does it sound right?

Reply:
It's more detailed. It's fine if only we don't add more other validations except for checking payment.total in ```is_fully_paid```.

Reply:
Maybe we could use same message as like in the comments: ```Provided payment methods can not cover the checkout's total amount.```